lib/Loop/Driver.php

```diff
@@ -62,20 +62,28 @@ abstract class Driver
      */
     public function run()
     {
-        $this->running = true;
+        $nesting = $this->nesting++;
 
         try {
-            while ($this->running) {
+            while ($nesting < $this->nesting) {
```

Comment:
Shouldn't this be `$this->isRunning()`?

Reply:
No. We want this particular while loop to end if the nesting level goes below the value when it was started. `$this->isRunning()` is true regardless of nesting depth.